### ISSUE TITLE
Create an "old base url" field for search

### ISSUE THREAD

**Issue by bondjimbond** [bondjimbond/old_url_lookup#7]
Recently removed the base URL from the Solr search string in [the last commit](https://github.com/bondjimbond/islandora_url_redirector/commit/86a06b9dbe4baae20776b833043b7d6e12b15da8). Should the base URL piece be added back in, making it configurable?

Possible use case: 
Currently, we're just looking for the URL snippet, whatever that might be (e.g. facultypub/*) instead of the full URL (e.g. http://kora.kpu.ca/facultypub/*). If an object's description for some reason contains the same string (unlikely but possible), we could end up returning the wrong object.

Instead, perhaps add another config form option: 
Old repository base URL - default ""
Possible issue: how to handle the presence or absence of slashes.

**Comment by mjordan** [comment]
@bondjimbond if the Solr queries are against the dc.identifier field, for example, how likely is this to happen?

**Comment by bondjimbond** [comment]
I'm not sure. Based on the previous line with the hard-coded URL, `request_path()` appears to only grab the bit after the base URL:

```
$path = request_path();
$path = "http://kora.kpu.ca/" . $path;
$pid = islandora_url_redirector_query_solr($path);
```
Current code drops the second line entirely.

I'd say it's pretty unlikely, however it's always possible. Generally best practice to avoid the possibility, no? 

Another possibility is that two repositories using similar structure get migrated to one new repository, and conflicts arise there. Again, not super likely, but possible.

I'm thinking one more variable in the admin from, something like this:

```
  $form['islandora_url_redirector_old_base'] = array(
    '#type' => 'textfield',
    '#title' => t('New hostname base URL'),
    '#size' => 60,
    '#default_value' => variable_get('islandora_url_redirector_old_base', ''),
    '#description' => t('Your old repository base URL'),
    '#maxlength' => 255,
  );

```
and
```
$path = request_path();
$path = variable_get'islandora_url_redirector_old_base' . $path;
$pid = islandora_url_redirector_query_solr($path);
```

If the user left the old_base field blank, it would stay blank, and the code would function as it currently does, just searching for the snippet. If they defined the old base URL, it would make sure that the complete URL was in the identifier and avoid potential conflicts.

..

What do you think, @mjordan ... worth doing, or too cumbersome?

**Comment by mjordan** [comment]
My $0.02 is it's too cumbersome.

First, `request_path()` only returns the part after the basename, so we do need to prepend the basename for queries.

Also, I'm having trouble wrapping my head around the use case of two repositories using a single instance of this module to do redirects. As far as I can imagine, the only way this would work is if each old repo's hostnames were aliased to the Drupal running the module. In which case, it would be possible to get more than one result from Solr if two objects' old URLs were  identical other than their respective hostnames, e.g., "http://example.org/path/to/item/100" and "http://example.edu/path/to/item/100". It would be difficult to disambiguate which object was the desired one by prepending a baseurl to the value returned by `request_path()`. For a given request, how would we know which basename to prepend?

Third, looking at the [Drupal API docs](https://api.drupal.org/api/drupal/includes!bootstrap.inc/function/request_path/7.x) for `request_path()`, we may have a different issue. The second example is:

```
http://example.com/drupalfolder/node/306 returns "node/306" while base_path() returns "/drupalfolder/".
```

If the Drupal running this module is installed in a subdirectory, `request_path()` is not going to return what we expect.

My advice is to not cover these edge cases ('cause that's what I think they are) for now and instead to document them under a "known issues" section in the README.

**Comment by bondjimbond** [comment]
> First, `request_path()` only returns the part after the basename, so we do need to prepend the basename for queries.

If you're saying that we DO need to prepend the basename for queries, then we DO need to make it configurable. If the new repository base URL is different from the old base URL, appending the current base URL will cause the query to fail. (And I did test this, accidentally, when I misunderstood how the line worked the issue you originally opened.)

Do you agree, then, that a config field is needed?

**Comment by mjordan** [comment]
Yes, I think it's safest to prepend the basename for queries, since that will yield the most precise and unambiguous results. The value of `global $basename` is a good default but as you point out that might not be sufficient. Therefore, the part that gets prepended should be configurable.

**Comment by bondjimbond** [comment]
Created a pull request #8 - does it meet the criteria?

**Comment by mjordan** [comment]
OK, I'll review this evening.

**Comment by bondjimbond** [comment]
Merged #8
